### PR TITLE
Add single resource admin concept

### PR DESCRIPTION
**Pull request by jacobherrington**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description

This PR is a layup based on some of the stuff @maestromac was working on regarding authorization for individual `internal` controllers.

We decided to rely on Rolify rather than writing something bespoke (which might be more elegant), to keep things consistent.

That being said, this is arguably a hack of what Rolify intends us to do -- because Rolify isn't really intended to be used to restrict views so much as access to models. We create empty model classes without DB-backing and call `resourcify` on them to use Rolify for limiting access to their controllers.

There is a strong argument to be made, in my opinion, that many of these views should be refactored into ad-hoc reports (because they are just AR queries spit out on a page). But for the time being, this allows us to divvy up access to internal resources.

One major limitation of this approach is that a user cannot be a "single resource admin" for multiple resources (afaik).

Based on manual testing and the automated tests that I wrote, this approach works. I'd like to expand the tests, but unfortunately writing tests for existing functionality that is touched by this I found pre-existing bugs that triggered test failures, so those will need to be addressed before additional specs can be written.

## Added to documentation?

Once this is merged, I would like to follow-up with a documentation PR to explain the concept of a single resource admin.

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

A cat knocking over a baby and baby gate

### CODE REVIEW

File: app/controllers/internal/application_controller.rb [hunk 1]
```diff
@@ -4,7 +4,11 @@ class Internal::ApplicationController < ApplicationController
 
   private
 
+  def authorization_resource
+    self.class.name.demodulize.sub("Controller", "").singularize.constantize
```
Comment (jacobherrington) [inline app/controllers/internal/application_controller.rb]: This feels fairly brittle and I'd like feedback, :grimacing:

**Comment by jacobherrington** [comment]
> A user can, in fact, be a single_resource_admin for multiple resources. So this part works.

I had trouble replicating this, but glad to hear you could do it. :sweat_smile: It might be worth adding some tests around that functionality.

File: app/controllers/internal/application_controller.rb (hunk 1 above)
Comment (rhymes) [reply app/controllers/internal/application_controller.rb]: AHHAHAHA this is a mouthful but glat it worked!

File: app/models/config.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Config < ApplicationRecord
```
Comment (rhymes) [inline app/models/config.rb]: @jacobherrington would it make sense to move these in an `models/internal` namespace/folder?

**Pull request merged**